app/src/main/java/com/project/squirrelobserver/squirrelObserver/RecordActivity.java

```diff
@@ -52,9 +54,26 @@ protected void onCreate(Bundle savedInstanceState) {
 
         //Setup timer
         mChronometer = (Chronometer) findViewById(R.id.timer);
+        modeToggle = (Switch) findViewById(R.id.modeToggle);
+        modeToggle.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Switch toggle = (Switch) v;
+                scanMode = toggle.isChecked();
+                if (toggle.isChecked()) {
+                    modeToggle.setText(R.string.record_activity_scan_label);
+                    actorTabActivity.switchFromAllOccurances();
```

Comment:
Use a null checker for accessing actorTabActivity, I implemented this really badly, so we should always check if it's null first.